src/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -9,18 +9,34 @@ namespace System.Diagnostics {
   public partial class Activity {
     public Activity(string operationName) {}      
     public string OperationName { get { throw null; } }
+
+    [System.ObsoleteAttribute]
     public string Id {get { throw null; } private set {} }
+
     public DateTime StartTimeUtc {get { throw null; } private set {} }
     public Activity Parent {get { throw null; } private set {} }
-    public string ParentId {get { throw null; } private set {} }
-    public string RootId {get { throw null; } private set {} }    
+  
+    public string ParentId {get { throw null; } set {} } // TraceId, public setter for binary (e.g. AMQP) or non-w3c custom propagation (e.g.Zipkin)
+
+    public string RootId {get { throw null; } set {} } // TraceId, public setter for binary (e.g. AMQP) or non-w3c custom propagation (e.g.Zipkin)
+
+    public string SpanId { get { throw null; } private set { } }
+
+    public byte TraceFlags {get { throw null; } set {} }
+
+    public string Tracestate {get { throw null; } set {} }
+
+    public string W3CId { get { throw null; } set { } }    // 00-traceid-spanid-01
```

Comment:
Having something that generates the W3CId format for the Activity's ID is reasonable.    Having the setter is not quite so clear.    If we have a Setter it should probably just be called SetId() and we infer the format from what it was given.  

However I don't really want to expose a setter for Id because it invites abuse.   You only seem to need it if you want to a 'pass through' activity.   I want to think about that case.    First, it does not seem that useful (when do you care enough about logging that you log everywhere by OUR system, but decide you don't want to log there). 

It feels like that case is best covered by an activity that has not been started (that is you did not log it).  

Anyway, it needs some thought, but I am biased against the setter.

Reply:
I have looked at how you use the W3CId setter below.   The thing is, that ID really is the parentID (indeed the HTTP field is traceparent, making it clear that it is the thing that 'caused' this request'.    Setting the parentID is completely reasonable (we have something that already does that).    Frankly I would like to just ignore the 'no start' case as mentioned above, but if we had to I would want to have something like 'TakeOnParentIdentity' which is used like Start, but basically just makes the current ID the same as the parents.